src/snowflake/snowpark/_internal/data_source_utils.py

```diff
@@ -247,27 +247,27 @@ def oracledb_to_snowpark_type(schema: List[Any]) -> StructType:
 
 
 def infer_data_source_schema(
-    conn: Connection, table: str, dbms_type: DBMS_TYPE, driver_info: str
+    conn: Connection, table_or_query: str, dbms_type: DBMS_TYPE, driver_info: str
 ) -> StructType:
     try:
         cursor = conn.cursor()
-        cursor.execute(f"SELECT * FROM {table} WHERE 1 = 0")
+        cursor.execute(f"SELECT * FROM {table_or_query} WHERE 1 = 0")
```

Comment:
`table_or_query` needs `()` for the subquery?

Reply:
as per discussion, sql server doesn't support `()` subquery